File: src/main/java/icu/samnyan/aqua/sega/allnet/model/Keychip.java
package icu.samnyan.aqua.sega.allnet.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity(name = "Keychip")
@Table(name = "allnet_keychips", uniqueConstraints = {@UniqueConstraint(columnNames = {"keychip_id"})})
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Keychip implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    private long id;

    private String keychip_id;
}

File: src/main/java/icu/samnyan/aqua/sega/allnet/Keychip.kt
package icu.samnyan.aqua.sega.allnet

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.io.Serializable

@Entity
@Table(name = "allnet_keychips")
class Keychip(
    @Id
    val id: Long = 0,

    @Column(unique = true, nullable = false)
    val keychipId: String = "",
) : Serializable {
    companion object {
        const val serialVersionUID = 1L
    }
}

@Repository("KeyChipRepository")
interface KeyChipRepo : JpaRepository<Keychip?, Long?> {
    fun existsByKeychipId(keychipId: String?): Boolean
}